src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -36,24 +41,41 @@ public class ParquetRowBuffer extends AbstractRowBuffer<ParquetChunkData> {
   private static final String PARQUET_MESSAGE_TYPE_NAME = "bdec";
 
   private final Map<String, Pair<ColumnMetadata, Integer>> fieldIndex;
+
+  /* map that contains metadata like typeinfo for columns and other information needed by the server scanner */
   private final Map<String, String> metadata;
+
+  /* Unflushed rows serialized into Java objects. Needed for the Parquet w/o memory optimization. */
   private final List<List<Object>> data;
+  /* BDEC Parquet writer. It is used to buffer unflushed data instead of serializing in Java objects */
+  private BdecParquetWriter bdecParquetWriter;
+
+  private ByteArrayOutputStream fileOutput;
   private final List<List<Object>> tempData;
+  private final String channelName;
 
   private MessageType schema;
-
+  private final List<List<Object>> testBuffer; // used only for tests for row index access
+  private final boolean bufferForTests;
+  private final boolean enableParquetMemoryOptimization;
   /** Construct a ParquetRowBuffer object. */
   ParquetRowBuffer(
       OpenChannelRequest.OnErrorOption onErrorOption,
       BufferAllocator allocator,
       String fullyQualifiedChannelName,
       Consumer<Float> rowSizeMetric,
-      ChannelRuntimeState channelRuntimeState) {
+      ChannelRuntimeState channelRuntimeState,
+      boolean bufferForTests,
+      boolean enableParquetMemoryOptimization) {
     super(onErrorOption, allocator, fullyQualifiedChannelName, rowSizeMetric, channelRuntimeState);
     fieldIndex = new HashMap<>();
     metadata = new HashMap<>();
     data = new ArrayList<>();
     tempData = new ArrayList<>();
+    channelName = fullyQualifiedChannelName;
+    testBuffer = new ArrayList<>();
```

Comment:
why to not just use `enableParquetMemoryOptimization = false` and `data` for tests? The code is the same for `data` and `testBuffer`, right?